src/Bladeburner/Actions/BlackOperation.ts

```diff
@@ -12,13 +13,22 @@ interface BlackOpParams {
 }
 
 export class BlackOperation extends ActionClass {
-  type: BladeburnerActionType.BlackOp = BladeburnerActionType.BlackOp;
-  name: BladeburnerBlackOpName;
+  readonly type: BladeburnerActionType.BlackOp = BladeburnerActionType.BlackOp;
+  readonly name: BladeburnerBlackOpName;
   n: number;
   reqdRank: number;
   teamCount = 0;
-  get id(): ActionIdentifier {
-    return { type: this.type, name: this.name };
+
+  get id() {
+    return BlackOperation.ActionIdentifier(this.name);
+  }
+
+  static ActionIdentifier(name: BladeburnerBlackOpName): TypeOfActionId<BlackOperation> {
```

Comment:
The name of this function is confusing, and it's also totally unneeded (it's only called once, immediately above here). Just inline this into get().

Reply:
It's used twice, which is why I pulled it out (here and the factory). I'll add a verb to it to help connect it.